framework/helpers/BaseArrayHelper.php

```diff
@@ -595,6 +595,9 @@ public static function getColumn($array, $name, $keepKeys = true)
      */
     public static function map($array, $from, $to, $group = null)
     {
+        if (is_string($from) && is_string($to) && $group === null) {
```

Comment:
Please also add a check that both `$from` and `$to` don't contain a dot (`.`).
The `ArrayHelper::getValue()` function supports getting a value by a "path" string.
Without this check the following code would break:
```php
$array = [
    [
        'prop' => [
            'a' => 'foo',
            'b' => 'bar',
        ],
    ],
];

ArrayHelper::map($array, 'prop.a', 'prop.b');
```

Reply:
Yes. It is necessary. @chriscpty would you please redo test with such a check?

Reply:
performance with the added check:

```
before: 1729233314.6667
after: 1729233315.4382
elapsed: 0.77146291732788
before: 1729233315.4382
after: 1729233316.245
elapsed: 0.80677795410156
before: 1729233316.245
after: 1729233317.0443
elapsed: 0.79932498931885
before: 1729233317.0443
after: 1729233317.8304
elapsed: 0.78613996505737
before: 1729233317.8305
after: 1729233318.6256
elapsed: 0.79515385627747
bench size: 18000
```

I'll add the fix (and related tests) in a second